pkg/gofr/auth.go

```diff
@@ -84,13 +84,22 @@ func (a *App) EnableAPIKeyAuthWithValidator(validateFunc func(c *container.Conta
 //
 // The JWKS endpoint is used to retrieve JSON Web Key Sets for verifying tokens.
 // The refresh interval specifies how often to refresh the token cache.
-func (a *App) EnableOAuth(jwksEndpoint string, refreshInterval int) {
+func (a *App) EnableOAuth(jwksEndpoint string,
+	refreshInterval int,
+	opts ...middleware.ClaimOption,
+) {
+	cnf := &middleware.ClaimConfig{}
+
+	for _, opt := range opts {
+		opt(cnf)
+	}
+
 	a.AddHTTPService("gofr_oauth", jwksEndpoint)
 
 	oauthOption := middleware.OauthConfigs{
 		Provider:        a.container.GetHTTPService("gofr_oauth"),
 		RefreshInterval: time.Second * time.Duration(refreshInterval),
 	}
 
-	a.httpServer.router.Use(middleware.OAuth(middleware.NewOAuth(oauthOption)))
+	a.httpServer.router.Use(middleware.OAuth(middleware.NewOAuth(oauthOption), cnf))
```

Comment:
My point is that you don't need the config here, you should pass the option to OAuth middleware. 

```go
func (a *App) EnableOAuth(jwksEndpoint string, 
	refreshInterval int,
	opts ...middleware.ClaimOption, 
 ) {
	a.AddHTTPService("gofr_oauth", jwksEndpoint)
	oauthOption := middleware.OauthConfigs{ 
		Provider:        a.container.GetHTTPService("gofr_oauth"), 
		RefreshInterval: time.Second * time.Duration(refreshInterval), 
	}

	a.httpServer.router.Use(middleware.OAuth(middleware.NewOAuth(oauthOption), opts...))
}
```

Then Oauth is like this

```go
func OAuth(key PublicKeyProvider, opts ClaimOption...) func(http.Handler) http.Handler {
	config := &ClaimConfig{} 
	for _, opt := range opts {
		opt(config) 
	}
```

This way App.EnableOAuth and  middleware.OAuth signature stays compatible